File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return None
        values = [head.val]
        pointer = head
        while pointer and pointer.next:
            if pointer.next.val in values:
                pointer.next = pointer.next.next
            else:
                values.append(pointer.next.val)
                pointer = pointer.next
        return head
                
        

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* cur = head;
            while(cur) {
                ListNode* tmp = cur;
                cur=cur->next;
                while(cur && tmp->val == cur->val) {
                    tmp->next=cur->next;
                    cur=cur->next;
                }
            }
        return head;
    }
};